### PR TITLE
Add ESLint Checks for the Repository

### PR DESCRIPTION
**Pull request by asekawa**
Add ESLint Checks for the Repository
- https://github.com/wso2-enterprise/asgardeo-product/issues/22992

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -27,7 +27,9 @@ module.exports = {
     extends: [
         "plugin:cypress/recommended",
         "eslint:recommended",
-        "plugin:import/typescript"
+        "plugin:import/typescript",
```
Comment (brionmario) [inline .eslintrc.js:30]: Is this working when you run the lint command? @asekawa 

```
plugins: ['@wso2'], // plugin should be added like this
  extends: [
    'plugin:@wso2/typescript',
    'plugin:@wso2/strict',
    'plugin:@wso2/internal',
    'plugin:@wso2/jest',
    'plugin:@wso2/prettier'
  ],
  ```

Comment (brionmario) [reply .eslintrc.js:30]: Also install, `eslint`, `prettier` & `jest` as dev dependencies

Comment (brionmario) [reply .eslintrc.js:30]: Test the lint command once you do all the changes.

Comment (asekawa) [reply .eslintrc.js]: These plugins were installed and tested

File: .eslintrc.js [hunk 2]
```diff
@@ -14,96 +14,27 @@
...
-    }
+  env: {
+    // other environments
+    jest: true,
+  },
+  extends: [
+    "plugin:@wso2/typescript",
+    "plugin:@wso2/strict",
+    "plugin:@wso2/internal",
+    "plugin:@wso2/prettier",
```
Comment (brionmario) [inline .eslintrc.js:30]:
```suggestion
    "plugin:@wso2/prettier",
    "plugin:cypress/recommended"
```

Comment (brionmario) [reply .eslintrc.js:30]: Create an issue in https://github.com/brionmario/wso2-ui-configs/issues repo to add a `@wso2/cypress` sharable config.

Comment (asekawa) [reply .eslintrc.js:30]: https://github.com/brionmario/wso2-ui-configs/issues/8

**Review by brionmario — APPROVED** [review]

**Pull request merged**